shm/Store.cpp

```diff
@@ -1,33 +1,152 @@
 #include "Store.hpp"
 
 #include <algorithm>
+#include <random>
 
-Store::Store(std::shared_ptr<Time>& time)
-    : time_(time) {
-    time_->addObserver(this);
+#include "Alcohol.hpp"
+#include "Fruit.hpp"
+#include "Item.hpp"
+
+Store::Store(std::shared_ptr<Time>& time) {
+    time_ = time;
+    time->addObserver(this);
+    market_ = makeStock(generateFruits(), generateAlcos(), generateItems());
+}
+
+size_t Store::generateRandom(int min, int max) const {
+    std::random_device rd;
+    std::mt19937 gen(rd());
+    std::uniform_int_distribution<> distirb(min, max);
+
+    return static_cast<size_t>(distirb(gen));
+}
+
+Cargos Store::generateFruits() const {
+    Cargos result;
+    result.reserve(marketSection);
+
+    size_t i = 0;
+    while(i < marketSection) {
+        Fruit fruit(fruitNames[generateRandom(0, 5)], generateRandom(1, 20), generateRandom(10, 30), generateRandom(1, 10));
+        if(std::none_of(begin(result), end(result),[&fruit](const auto& ptr){ return ptr->getName() == fruit.getName();})) {
+            result.emplace_back(std::make_shared<Fruit>(fruit));
+            i++;
+        }
+    }
```

Comment:
Is it possible that we would get only one fruit after this code will execute? I mean, if it happens that each time we get the same fruitName as the first one, then we won't add any more fruit to the Stock. Am I right?

Reply:
We won't add same fruit again but the `i` won't increament aswell so the loop continues untill we get new name

Reply:
Yeah, you're right. My bad.